ckanext/dataset_subscriptions/actions.py

```diff
@@ -39,8 +42,13 @@ def dms_notification_provider(user_dict, since):
                      if activity['user_id'] != user_dict['id']
                      and 'package' in activity['activity_type']]
     # We want a notification per changed dataset, not a list of all changes
-    deduplicated_activity_list = list({item["object_id"]: item for item in activity_list}.values())
+    deduplicated_activity_list = list({item["object_id"]: item for item in sorted(activity_list, key = lambda item: item['timestamp'])}.values())
     updated_activity_list = _add_dataset_name_to_activity_list(deduplicated_activity_list, context)
+    # Filter out the old activities.
+    strptime = datetime.datetime.strptime
+    fmt = '%Y-%m-%dT%H:%M:%S.%f'
+    updated_activity_list = [activity for activity in updated_activity_list
+            if strptime(activity['timestamp'], fmt) > since]
```

Comment:
this piece should be a separate function. Then you can refactor the comment into a function callout.

Reply:
done